niworkflows/utils/spaces.py

```diff
@@ -0,0 +1,114 @@
+"""Utilities for tracking and filtering spaces"""
+
+FSAVERAGE_DENSITY = {
+    '642': 'fsaverage3',
+    '2562': 'fsaverage4',
+    '10k': 'fsaverage5',
+    '41k': 'fsaverage6',
+    '164k': 'fsaverage',
+}
+NONSTANDARD_REFERENCES = ['anat', 'T1w', 'run', 'func', 'sbref', 'session', 'fsnative']
+
+
+class Spaces:
+    """Helper class for managing fMRIPrep spaces."""
+    __slots__ = ('_output', '_internal')
+
+    def __init__(self, output=None):
+        """
+        Tracks and distinguishes internal and output spaces.
+
+        Internal spaces are normalizations required for pipeline execution which
+        can vary based on user arguments.
+        Output spaces are desired user outputs.
+        """
+        self._output = []
```

Comment:
In reality this represents template specs, as opposed to the "internal" - which are actually spaces in BIDS terms. Please note the thin line between template and space.

I guess I don't love the nomenclature of internal/outputs for this new data structure.

Reply:
definitely open to suggestions here